src/editor/controls/Sidebar.tsx

```diff
@@ -390,6 +391,28 @@ export const SidebarControls = ({
                     />
                 </BaseControl>
             </PanelBody>
+            <PanelBody
+                title={__('GitHub', 'code-block-pro')}
+                initialOpen={bringAttention === 'language-select'}>
+                <div className="code-block-pro-editor">
+                    <BaseControl id="code-block-pro-language">
+                        <GitHubRepositoryControl
+                            value={attributes.url}
+                            onChange={(url) => {
+                                setAttributes({ url });
+                                updateThemeHistory({ url });
+                            }}
+                            onCodeFetched={({ code, lineNumbers }) => {
+                                setAttributes({
+                                    code,
+                                    enableHighlighting: Boolean(lineNumbers),
+                                    lineHighlights: Boolean(lineNumbers) ? `[${lineNumbers.startLine},${lineNumbers.endLine}]` : '',
+                                });
```

Comment:
I believe this should only be updating the `code` right?

Reply:
So those other lines are there to allow users to be able to preserve highlighted lines from GitHub links. Would you like me to remove that capability from this PR?

Reply:
Oh I didn't realize that. That's interesting. I like the idea actually.

I think that makes sense for the first time it fetches, but not for syncing long term though

Could it work like this?

1. User pastes the link
2. We fetch the code
3. Parse it to get line numbers
4. Remove the line numbers from the URL
5. Add line number highlights and URL (which no longer includes line numbers reference)
6. Then when syncing next time, next week, etc we just sync the code with no line numbers

So we just have it as is, but we dont persist the url with the line numbers. that way it only ever adds the highlights once. I'm just thinking about users that want to sync with a gist they edit, and the line numbers might now always be the same. i'd hate for them to need to add the URL.

Alternatively we could add a checkbox like "preserve line highlights" or something. 

What do you think?